packages/dashboard-frontend/src/services/oauth/index.ts

```diff
@@ -63,6 +63,7 @@ export default class OAuthService {
           redirectUrl.toString(),
         );
       }
+      throw e;
```

Comment:
@vinokurig @olexii4 folks, I'm a bit lost with the flow here:

- we are trying to refresh the token
- if the token is experied and we get 401, user is redirected to the OAuth auntefication page and after the redirect we throw an exception

How it used to work before, and why do we need to throw exception? Would it have some side-effects?

Reply:
> why do we need to throw exception?

We need the exception to prevent the workspace from starting before accepting the SCM authorisation.
When dashboard opens the SCM authentication page: 
https://github.com/eclipse-che/che-dashboard/blob/c2295c4eb18d26541b27ce3b35faf8a613ec336e/packages/dashboard-frontend/src/services/oauth/index.ts#L61-L64 
the code continues to execute some time, even after the page reload command. This causes to the workspace start,  even without authorising the authentication page.
When user authorises the SCM authentication page, the SCM provider requests the callback URL and the dashboard page loads again. During the second workspace start request, we check the workspace state before requesting the refresh token API. If the workspace state is `started` the token refresh request is skipped:
https://github.com/eclipse-che/che-dashboard/blob/c2295c4eb18d26541b27ce3b35faf8a613ec336e/packages/dashboard-frontend/src/store/Workspaces/devWorkspaces/index.ts#L291-L294
We need to throw the exception to prevent the workspace from starting, if an unauthorised error occurs.
>  How it used to work before

This bug is reproduced only with specific conditions:
* Workspace is created and started at least once.
* SCM authorisation is revoked.
* The OAuth token is removed or expired.

So looks like we have not tested this case before.
> Would it have some side-effects?

I don't think so because we just pass the error from the refresh token function further, which looks logical.